src/services/firebase.ts

```diff
@@ -115,45 +117,54 @@ export namespace ourFirebase {
     // Make sure we listen to match changes only once.
     // 'gamePortal/matches'
     // store.dispatch(updateMatchList);
-    let tempMatchIds: Promise<MatchInfo>[] = [];
+    let tempMatchIds: Promise<any>[] = [];
     for (let matchId of matchIds) {
       // getMatchDetail(matchId).then(() => {tempMatchIds.push(getMatchDetail(matchId))});
       const match = getMatchDetail(matchId);
+
       match.then(data => {
         if (data.status === 'resolved') {
-          tempMatchIds.push(getMatchDetail(matchId));
+          // tempMatchIds.push(getMatchDetail(matchId));
+          tempMatchIds.push(match);
         }
-        console.log(matchId);
       });
     }
-    Promise.all(tempMatchIds).then((datas: any) => {
```

Comment:
I test my code line by line, and find that the data is affected by some other test, so I  disabled one of the test.

And now I get stuck on this problem:

The data in tempMatchIds is valid here, but I can't read the value of datas, console.log(datas) shows nothing. 
And what confuses me most is that both the console.log('In the pormises' ) and console.log('Wrong when fetch matches') got printed, which means the content in then and catch are both executed.